controllers/node_kernel_reconciler.go

```diff
@@ -37,7 +38,7 @@ func (r *NodeKernelReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, fmt.Errorf("could not get node: %v", err)
 	}
 
-	kernelVersion := node.Status.NodeInfo.KernelVersion
+	kernelVersion := strings.TrimSuffix(node.Status.NodeInfo.KernelVersion,"+")
```

Comment:
```suggestion
	kernelVersion := strings.TrimSuffix(node.Status.NodeInfo.KernelVersion, "+")
```
We also need a unit test for this feature.

Reply:
Wondering if this behaviour should be optional. @ybettan @yevgeny-shnaidman thoughts?

Reply:
@qbarrand why do you think it should be optional? In case it is a valid kernel version string, it won't be changed. What use case do you see for keeping the "+" at the end of the kernel version?

Reply:
I do not see any, really. I am just thinking that customers might be adding patches that would be required by their kmod. But then those customers would probably be managing their kernel releases like RHEL does.